### PR TITLE
Adds a snippet for our recommended best practice around creating modules in JS

### PR DESCRIPTION
**Pull request by jansepar**
Adds a snippet for our recommended best practice around creating modules in JS

Status: **Opened for review**

Reviewers: Engineers @ Mobify

## Changes
- Defines our best practice around defining JS modules.

### CODE REVIEW

File: javascript/README.md [hunk 1]
```diff
@@ -505,3 +505,32 @@ functionThatReturnsAPromise()
         // other code here!
     });
 ```
+
+## Module Definition
+
+When building a module that can be consumed by a number of different sources, we prefer
```
Comment (jeremywiebe) [inline javascript/README.md]: What about modules that aren't consumed by a number of different sources... do we support a different module definition then?

Comment (jansepar) [inline javascript/README.md]: Nah - if it's just something your consuming that will only ever live in one project, it doesn't need to be UMD.

Comment (jansepar) [inline javascript/README.md]: I'll make that more clear.

File: javascript/README.md [hunk 2]
```diff
@@ -505,3 +505,36 @@ functionThatReturnsAPromise()
...
+to use [Universal Module Definition (UMD)](https://github.com/umdjs/umd), so that the
+module will be compatible with AMD, CommonJS, or plain script inclusion.
+For modules that are not shared across projects, UMD is not required.
```
Comment (mikenikles) [inline javascript/README.md]: @jeremywiebe and @jansepar: Do you think that covers your discussion above? If so, let's get this merged.

Comment (jansepar) [inline javascript/README.md]: Yup looks good.

**Pull request merged**